test/setToken.spec.ts

```diff
@@ -114,42 +112,28 @@ contract("SetToken", (accounts) => {
     });
   };
 
-  const deployStandardSetAndApprove = async (numComponents: number, customUnits: BigNumber[] = []) => {
-    await resetAndDeployComponents(numComponents, customUnits);
-
-    const setTokenTruffle = await SetToken.new(
-      componentAddresses,
-      units,
-      STANDARD_NATURAL_UNIT,
-      TX_DEFAULTS,
-    );
-
-    const setTokenWeb3Contract = web3.eth
-      .contract(setTokenTruffle.abi)
-      .at(setTokenTruffle.address);
-
-    setToken = new SetTokenContract(setTokenWeb3Contract, TX_DEFAULTS);
-
-    const approvePromises = _.map(components, (component) =>
-      component.approve.sendTransactionAsync(setToken.address, UNLIMITED_ALLOWANCE_IN_BASE_UNITS, TX_DEFAULTS),
-    );
-
-    await Promise.all(approvePromises);
-  };
-
   describe("Creation", async () => {
+    let caller: Address = testAccount;
+    
+    async function subject(): Promise<any> {
+      return SetToken.new(
+        componentAddresses,
+        units,
+        naturalUnit,
+        name,
+        symbol,
+        { from: caller },
+      );
+    }
+
     describe(`of Standard Set`, () => {
       beforeEach(async () => {
         await resetAndDeployComponents(2);
       });
 
       it("should work with the correct data", async () => {
-        const setTokenTruffleInstance = await SetToken.new(
-          _.map(components, (component) => component.address),
-          units,
-          STANDARD_NATURAL_UNIT,
-          TX_DEFAULTS,
-        );
+
+        const setTokenTruffleInstance = await subject();
```

Comment:
This setup, along with setup directly below it should go into coreWrapper (or some other util). 

```
const setTokenWeb3Instance = web3.eth
             .contract(setTokenTruffleInstance.abi)
             .at(setTokenTruffleInstance.address);
 
const setTokenInstance: SetTokenContract = new SetTokenContract(setTokenWeb3Instance, TX_DEFAULTS);
```

Reply:
The subject can just call `coreWrapper.deploySetTokenAsync(...)` and toggle those for nested test cases.